src/main/java/org/openrewrite/java/testing/junit5/AddParameterizedTestAnnotation.java

```diff
@@ -0,0 +1,74 @@
+package org.openrewrite.java.testing.junit5;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.tree.J;
+
+public class AddParameterizedTestAnnotation extends Recipe {
+    private static final AnnotationMatcher TEST_ANNOTATION_MATCHER = new AnnotationMatcher("@org.junit.jupiter.api.Test");
+    private static final AnnotationMatcher VALUE_SOURCE_ANNOTATION_MATCHER = new AnnotationMatcher("@org.junit.jupiter.params.provider.ValueSource");
```

Comment:
Had you noticed that this matcher takes in a pattern that can include parameters as well? That might be why your matcher fails. Perhaps something to trace with the debugger.

Reply:
How should I update the pattern? I kinda know what needs to be added but I'm not sure how to do it in a way thats generic enough? Would something like ValueSource("#{}") work?